scripts/triage_failures.py

```
@@ -97,6 +98,42 @@ def find_failed_step(job):
+)
+
+
+def run_gh(args, timeout=None, attempts=4, delay=2):
+    """Run a `gh` command, retrying transient API failures with backoff.
+
+    Returns the CompletedProcess of the final attempt; callers check
+    ``returncode`` themselves. Propagates FileNotFoundError when gh is missing.
+    """
+    result = None
+    for attempt in range(1, attempts + 1):
+        result = subprocess.run(["gh"] + args, capture_output=True, text=True,
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)